File: exercises/py101/easy2_small_problems/middle_char.py
def center_of_even_sentence(sentence):
    # launch len(sentence) // 2 = 3 - 1
    left_index = len(sentence) // 2 - 1
    return sentence[left_index:left_index + 2]

def center_of_odd_sentence(sentence):
    # odd len(sentence) // 2 = 1
    center_index = len(sentence) // 2 
    return sentence[center_index]

def center_of(sentence):
    # assume len(sentence) > 0 
    if len(sentence) % 2 == 1:
        return center_of_odd_sentence(sentence)
    else:
        return center_of_even_sentence(sentence)

print(center_of('I Love Python!!!') == "Py")    # True
print(center_of('Launch School') == " ")        # True
print(center_of('Launchschool') == "hs")        # True
print(center_of('Launch') == "un")              # True
print(center_of('Launch School is #1') == "h")  # True
print(center_of('x') == "x")                    # True

File: exercises/LS215/easy5/middle-char.js

/*
'x' => 'x' => idx 1
'Launch' => 'un' idx 2, 3
'Launch School' => ' ' idx 6
Math.floor(str.length / 2)

odd length:  return one character
even length: return two characters
*/
const isEvenLength = (string) => string.length % 2 === 0;

function centerOf(string) {
  let middleIdx = Math.floor(string.length / 2);

  if (isEvenLength(string)) {
    return string.slice(middleIdx - 1, middleIdx + 1);
  } else {
    return string[middleIdx];
  }
}


console.log(centerOf('I Love JavaScript')); // "a"
console.log(centerOf('Launch School'));     // " "
console.log(centerOf('Launch'));            // "un"
console.log(centerOf('Launchschool'));      // "hs"
console.log(centerOf('x'));                 // "x"